spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/DnnGraph.scala

```diff
@@ -409,30 +422,66 @@ class DnnGraph(
     }
   }
 
+  private def getHeapFormat(inputs: Array[MemoryData]): Int = {
+    var heapFormat: Int = -1
+    inputs.foreach(m => {
+      if (m.shape.length == 4) {
+        return inputs(0).layout
+      }
+    })
+
+    @inline
+    def transferFormat(format: DataFormat): Int = {
+      if (format == DataFormat.NHWC) Memory.Format.nhwc else Memory.Format.nchw
```

Comment:
Does it mean we have no `ntc` and `tnc`.

Reply:
For 3 dims, Blas layer only supports `ntc` , so it's no need to distinguish `ntc` and `tnc`.
But for 4 dims, there are two formats `nchw` and `nhwc` in some layers, so I have to iterate through all layers to find the real format.